pkg/os/volume/api.go

```diff
@@ -149,92 +177,117 @@ func (VolumeAPI) UnmountVolume(volumeID, path string) error {
-	cmdEnv = append(cmdEnv, fmt.Sprintf("volumeID=%s", volumeID))
-	cmdEnv = append(cmdEnv, fmt.Sprintf("mountpath=%s", path))
-	out, err := utils.RunPowershellCmd(cmd, cmdEnv...)
-
+	mountPoint := path
+	if !strings.HasSuffix(mountPoint, "\\") {
+		mountPoint += "\\"
+	}
+	utf16MountPath, _ := windows.UTF16PtrFromString(mountPoint)
+	err := windows.DeleteVolumeMountPoint(utf16MountPath)
 	if err != nil {
-		return fmt.Errorf("error getting driver letter to mount volume. cmd: %s, output: %s,error: %v", cmd, string(out), err)
+		return fmt.Errorf("error umount volume (%s) from path %s. error: %v", volumeID, path, err)
 	}
 	return nil
 }
 
 // ResizeVolume - resizes a volume with the given size, if size == 0 then max supported size is used
 func (VolumeAPI) ResizeVolume(volumeID string, size int64) error {
-	// If size is 0 then we will resize to the maximum size possible, otherwise just resize to size
-	var cmd string
-	var out []byte
 	var err error
 	var finalSize int64
-	var outString string
-	if size == 0 {
-		cmd = `Get-Volume -UniqueId "$Env:volumeID" | Get-partition | Get-PartitionSupportedSize | Select SizeMax | ConvertTo-Json`
-		cmdEnv := fmt.Sprintf("volumeID=%s", volumeID)
-		out, err := utils.RunPowershellCmd(cmd, cmdEnv)
+	part, err := cim.GetPartitionByVolumeUniqueID(volumeID, nil)
+	if err != nil {
+		return err
+	}
 
-		if err != nil || len(out) == 0 {
-			return fmt.Errorf("error getting sizemin,sizemax from mount. cmd: %s, output: %s, error: %v", cmd, string(out), err)
+	// If size is 0 then we will resize to the maximum size possible, otherwise just resize to size
+	if size == 0 {
+		var sizeMin, sizeMax ole.VARIANT
+		var status string
+		result, err := part.InvokeMethodWithReturn("GetSupportedSize", &sizeMin, &sizeMax, &status)
+		if result != 0 || err != nil {
+			return fmt.Errorf("error getting sizemin, sizemax from volume (%s). result: %d, error: %v", volumeID, result, err)
 		}
 
-		var getVolumeSizing map[string]int64
-		outString = string(out)
-		err = json.Unmarshal([]byte(outString), &getVolumeSizing)
+		finalSizeStr := sizeMax.ToString()
+		finalSize, err = strconv.ParseInt(finalSizeStr, 10, 64)
 		if err != nil {
-			return fmt.Errorf("out %v outstring %v err %v", out, outString, err)
+			return fmt.Errorf("error parsing the sizeMax of volume (%s) with error (%v)", volumeID, err)
 		}
-
-		sizeMax := getVolumeSizing["SizeMax"]
-
-		finalSize = sizeMax
 	} else {
 		finalSize = size
 	}
 
-	currentSize, err := getVolumeSize(volumeID)
+	currentSizeVal, err := part.GetProperty("Size")
 	if err != nil {
 		return fmt.Errorf("error getting the current size of volume (%s) with error (%v)", volumeID, err)
 	}
 
+	currentSize, err := strconv.ParseInt(currentSizeVal.(string), 10, 64)
+	if err != nil {
+		return fmt.Errorf("error parsing the current size of volume (%s) with error (%v)", volumeID, err)
+	}
+
 	// only resize if finalSize - currentSize is greater than 100MB
 	if finalSize-currentSize < 100*1024*1024 {
```

Comment:
```
	if finalSize-currentSize < 100*1024*1024 {
		klog.V(2).Infof("minimum resize difference(1GB) not met, skipping resize. volumeID=%s currentSize=%d finalSize=%d", volumeID, currentSize, finalSize)
		return nil
	}
```

Reply:
just add `return nil` in this condition, that's one regression I have been found in our disk csi driver e2e test failures.

Reply:
Thanks for catching it.
I believe it was a mistake during a local merge with another commit.
I'll push a new commit to fix it.